Add beta documentation for SQL Server (#12219)
DataDog/documentation: content/en/database_monitoring/setup_sql_server/selfhosted.md

2. Copy the `odbc.ini` and `odbcinst.ini` files into the `/opt/datadog-agent/embedded/etc` folder.

## Grant the Agent access

The Datadog Agent requires read-only access to the database server in order to collect statistics and queries.

Create a read-only login to connect to your server:

```SQL
CREATE LOGIN datadog WITH PASSWORD = '<PASSWORD>';
CREATE USER datadog FOR LOGIN datadog;
GRANT SELECT on sys.dm_os_performance_counters to datadog;
GRANT VIEW SERVER STATE to datadog;
```

To collect file size metrics per database, ensure the user you created (`datadog`) has [connect permission][5] access to your databases by running:

```SQL
GRANT CONNECT ANY DATABASE to datadog;
```

Additionally to collect AlwaysOn and `sys.master_files` metrics, grant the following additional permission:

```SQL
GRANT VIEW ANY DEFINITION to datadog;
```

## Install the Agent

If you haven't already installed the Agent for your SQL Server database host, see the [Agent installation instructions][6].

1. Edit the Agent's `conf.d/sqlserver.d/conf.yaml` file to point to your `host` / `port` and set the hosts to monitor. See the [sample sqlserver.d/conf.yaml][7] for all available configuration options.

> {{< tabs >}}
> {{% tab "Windows" %}}

Reviewer: In this instance, the tabs are breaking, and this is a very small blurb from each that I think we can display both at once without the tabs, especially since we haven't split them up earlier on in the page when there were extra steps for linux.

```suggestion
   **Windows:**
```

    ```yaml
    init_config:
    instances:
      - dbm: true
        host: '<SQL_HOST>,<SQL_PORT>'
        username: datadog
        password: '<PASSWORD>'
        connector: odbc  # Alternative is 'adodbapi'
        driver: SQL Server
        tags:  # optional
          - 'service:<CUSTOM_SERVICE>'
          - 'env:<CUSTOM_ENV>'
    ```

> {{% /tab %}}
> {{% tab "Linux" %}}

Reviewer: ```suggestion
   **Linux**:
```

>     For SQL Server installations running on Linux hosts, use the `odbc` connector and specify the proper driver as indicated in the `odbcinst.ini` file.

Reviewer: ```suggestion
   For SQL Server installations running on Linux hosts, use the `odbc` connector and specify the proper driver as indicated in the `odbcinst.ini` file.
```

    ```yaml
    init_config:
    instances:
      - dbm: true
        host: '<SQL_HOST>,<SQL_PORT>'
        username: datadog
        password: '<PASSWORD>'
        connector: odbc
        driver: '<Driver from the `odbcinst.ini` file>'
        tags:  # optional
          - 'service:<CUSTOM_SERVICE>'
          - 'env:<CUSTOM_ENV>'
    ```

> {{% /tab %}}
> {{< /tabs >}}

Reviewer: ```suggestion

```

>   The `service` and `env` tags allow you to link your database telemetry to other telemetry through a common tagging scheme. To see how these tags are used throughout Datadog, see the documentation on [unified service tagging][8].

Reviewer: ```suggestion
   The `service` and `env` tags allow you to link your database telemetry to other telemetry through a common tagging scheme. To learn how these tags are used throughout Datadog, read the documentation on [unified service tagging][8].
```

>   See the [example check configuration][7] for a comprehensive description of all options, including how to use custom SQL queries to create your own metrics.

Reviewer: ```suggestion
   See the [example check configuration][7] for a comprehensive description of all options, including how to use custom SQL queries to create your own metrics.
```

>   **Note**: The (default) provider `SQLOLEDB` is being deprecated. To use the newer `MSOLEDBSQL` provider, set the `adoprovider` variable to `MSOLEDBSQL` in your `sqlserver.d/conf.yaml` file after having downloaded the new provider from [Microsoft][9]. It is also possible to use the Windows Authentication and not specify the username/password with:

Reviewer: ```suggestion
   **Note**: The (default) provider `SQLOLEDB` is deprecated and no longer maintained. To use the newer `MSOLEDBSQL` provider, set the `adoprovider` variable to `MSOLEDBSQL` in your `sqlserver.d/conf.yaml` file after having downloaded the new provider from [Microsoft][9]. It is also possible to use the Windows Authentication and not specify the username and password with:
```

  ```yaml
  connection_string: "Trusted_Connection=yes"

>   ```

Reviewer: Just trying to make sure everything up to this point is indented/included in step one with three spaces.
~~~suggestion
   ```yaml
   connection_string: "Trusted_Connection=yes"
   ```
~~~

2. [Restart the Agent][10].

### Collecting logs (optional)

> 1. Collecting logs is disabled by default in the Datadog Agent, enable it in your `datadog.yaml` file:

Reviewer: ```suggestion
1. Collecting logs is disabled by default in the Datadog Agent. To enable it, set `logs_enabled: true` in your `datadog.yaml` file:
```

    ```yaml
    logs_enabled: true
    ```

> 2. Add this configuration block to your `sqlserver.d/conf.yaml` file to start collecting your SQL Server logs:

Reviewer: ```suggestion
2. To start collecting your SQL Server logs, add this configuration block to your `sqlserver.d/conf.yaml` file:
```

    ```yaml
    logs:
      - type: file
        encoding: utf-16-le
        path: "<LOG_FILE_PATH>"
        source: sqlserver
        service: "<SERVICE_NAME>"
    ```

    Change the `path` parameter value to be the file path of your SQL Server log file.

>     The `service` parameter allows you to tie your logs to other telemetry through a common `service` tag. To see how `service` is used throughout Datadog, see the documentation on [unified service tagging][8].

Reviewer: ```suggestion
    The `service` parameter allows you to tie your logs to other telemetry through a common `service` tag. To learn more about how `service` is used throughout Datadog, read the documentation on [unified service tagging][8].
```

3. [Restart the Agent][10].

### Validate

> [Run the Agent's status subcommand][11] and look for `sqlserver` under the Checks section. Or visit the [Databases][12] page to get started!

Reviewer: ```suggestion
[Run the Agent's status subcommand][11] and look for `sqlserver` under the **Checks** section. Or navigate to the [Databases][12] page in Datadog to get started!
```

## Troubleshooting

> If you have installed and configured the integrations and Agent as described and it is not working as expected, see [Troubleshooting][13]

Reviewer: ```suggestion
If you have installed and configured the integrations and Agent as described and it is not working as expected, read about [Troubleshooting Database Monitoring][13].
```

## Further reading

{{< partial name="whats-next/whats-next.html" >}}

[1]: /agent/basic_agent_usage#agent-overhead
[2]: /database_monitoring/data_collected/#sensitive-information
[3]: https://docs.microsoft.com/en-us/sql/tools/configuration-manager/tcp-ip-properties-ip-addresses-tab
[4]: https://docs.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server?view=sql-server-2017
[5]: https://docs.microsoft.com/en-us/sql/t-sql/statements/grant-server-permissions-transact-sql?view=sql-server-ver15
[6]: https://app.datadoghq.com/account/settings#agent
[7]: https://github.com/DataDog/integrations-core/blob/master/sqlserver/datadog_checks/sqlserver/data/conf.yaml.example
[8]: /getting_started/tagging/unified_service_tagging
[9]: https://docs.microsoft.com/en-us/sql/connect/oledb/oledb-driver-for-sql-server?view=sql-server-2017
[10]: /agent/guide/agent-commands/#start-stop-and-restart-the-agent
[11]: /agent/guide/agent-commands/#agent-status-and-information
[12]: https://app.datadoghq.com/databases

> [13]: /database_monitoring/troubleshooting/?tab=sqlserver

Reviewer: There isn't a sqlserver tab on this page yet, is that coming in another PR?

Author: This will be added sometime in the future. Can we leave it like this so we don't forget to update it later? It should not break anything.
